3-enrich/stream-enrich/integration/src/test/scala/com.snowplowanalytics.snowplow.enrich.stream/KafkaIntegrationSpec.scala

```diff
@@ -0,0 +1,143 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the Apache License Version 2.0 is distributed on an "AS
+ * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics
+package snowplow
+package enrich
+package stream
+
+// Scala
+import com.snowplowanalytics.snowplow.enrich.stream.model.StreamsConfig
+
+import scala.concurrent._
+import scala.concurrent.duration.Duration
+import scala.util.Try
+import collection.JavaConversions._
+import scala.concurrent.forkjoin.ForkJoinPool
+
+// Java
+import java.util.Properties
+
+// Scala libraries
+import org.apache.kafka.clients.consumer.{ConsumerRecords, KafkaConsumer}
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+import io.bfil.kafka.specs2.DefaultKafkaPorts
+import io.bfil.kafka.specs2.EmbeddedKafkaContext
+
+// Specs2
+import org.specs2.matcher.{TraversableMatchers, TryMatchers}
+
+// Snowplow and Iglu
+import scalatracker.Tracker
+import iglu.client.Resolver
+import common.enrichments.EnrichmentRegistry
+
+/*
+ * Extending this trait creates a new integration test with a new instance of kafka
+ * See PiiEmitSpec for an example of how to use it
+ */
+trait KafkaIntegrationSpec
+    extends EmbeddedKafkaContext
+    with DefaultKafkaPorts
+    with TryMatchers
+    with TraversableMatchers {
+
+  import KafkaIntegrationSpecValues._
+  implicit val ec = ExecutionContext.fromExecutor(new ForkJoinPool(16))
+  val kafkaTopics = Set(testGoodIn, testGood, testBad, testPii)
+
+  def expectedGood: Int
+  def expectedBad: Int
+  def expectedPii: Int
+
+  def inputGood: List[Array[Byte]]
+
+  def getMainApplicationFuture(
+                                configuration: StreamsConfig,
+                                resolver: Resolver,
+                                registry: EnrichmentRegistry,
+                                tracker: Option[Tracker]): Future[Unit] = Future {
+    KafkaEnrich.getSource(configuration, resolver, registry, tracker).toOption.get.run()
+  }
+
+  def producerTimeoutSec: Int
+  def inputProduced: Try[Unit] = Try { Await.result(produce, Duration(s"$producerTimeoutSec sec")) }
+  def produce: Future[Unit] = Future {
+    val testKafkaPropertiesProducer = {
+      val props = new Properties()
+      props.put("bootstrap.servers", kafkaHost)
+      props.put("client.id", "producer-george")
+      props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
+      props.put("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+      props
+    }
+
+    val testProducer = new KafkaProducer[String, Array[Byte]](testKafkaPropertiesProducer)
+    val events = inputGood
+    events.foreach { r =>
+      testProducer.send(new ProducerRecord(testGoodIn, "key", r))
```

Comment:
why are you using a future if you're discarding the result of the only async code here?

Reply:
There are futures that have only some effect and don't return anything..